### DISCUSSION TITLE
0.55.1 to latest, challenge with bindings in view

### DISCUSSION THREAD

**Discussion by imfractured** [pointfreeco/swift-composable-architecture#2966] [Q&A] [answered]
So in my legacy TCA app currently we use Bool binding type in the `get` that is a closure to determine the true / false value based on a more complex logic, ie in this case if the status enum type is a certain enum value. I'm trying to migrate to the latest version with the @Bindable and the new structure seems something like $store.status.sending(\.hideBaggingView) etc. I'm struggling to understand how the binding action / binding reducer can be used to solve this same issue? or maybe im going down the wrong path. Any help would be much appreciated, small stumbling block that I've spent quite a few hours trying to wrap my brain around.

`.fullScreenCover(
                        isPresented: viewStore.binding(
                            get: { $0.status == .bagging },
                            send: BaggingListAction.hideBaggingView
                        )
                    ) `

**Comment by stephencelis** [comment, marked as answer]
@imfractured That's a big upgrade! Did you go one step at a time and follow the migration guides?

As for how to create an ad hoc binding, the modern equivalent is documented here: https://pointfreeco.github.io/swift-composable-architecture/main/documentation/composablearchitecture/bindings/#Ad-hoc-bindings

In your case you have a lot of computation happening, so the best you can do to approximate this is with a custom property on your store:

```swift
extension StoreOf<MyFeatureName> {
  var isBaggingPresented: Bool {
    get { status == .bagging }
    set {
      if !newValue {
        send(.hideBaggingView)
      }
    }
  }
}

// ...

.fullScreenCover(isPresented: $store.isBaggingPresented) { /* ... */ } 
```

**Reply by imfractured** [reply]
hey @stephencelis thanks so much for the super quick reply! Been using TCA since 0.3x and our team absolutely loves it! I was hoping to avoid the extra extension or computed properties but in this case it seems like the best way to keep the store logic out of the view and more testable. Really loving the new ergonomics but we have a lot of work to do to get fully on board with the latest. Keep up the great work <3

**Comment by mbrandonw** [comment]
Hi @imfractured, you should be able add an `isBagging` computed property to `Status` (or to your `State` that holds `Status`), and update the action to something like `case setBagging(Bool)` to take a boolean, and then you can do something like this in the view:

```swift
$store.status.isBagging.sending(\.setBagging)
```

Also I will mention that we highly recommend you upgrade your app one version at a time (i.e. first to 1.0, then 1.1, then 1.2, etc.). That makes it far easier to find problems without needing to understand all versions of TCA at once.

**Reply by imfractured** [reply]
hi @mbrandonw thanks a million for the quick reply, will look at making these changes and changing my habits with the latest version, so much amazing goodness to take in, but you're right gotta take baby steps here, just getting excited for all the amazing changes, you guys rock, please keep up the great work!!
